monkey/monkey_island/cc/models/monkey.py

```diff
@@ -95,6 +100,18 @@ def is_dead(self):
                 monkey_is_dead = True
         return monkey_is_dead
 
+    def has_parent(self):
+        for p in self.parent:
+            if p[0] != self.guid:
+                return True
+        return False
+
+    def get_parent(self):
+        if self.has_parent():
+            Monkey.objects(guid=self.parent[0][0]).first()
+        else:
+            raise ParentNotFoundError
```

Comment:
We should add  some context to this error.

```suggestion
            raise ParentNotFoundError(f"No parent was found for agent with GUID {self.guid}")
```